api/models/User.js

```diff
@@ -31,4 +40,155 @@ userSchema.statics.create = function(name) {
 	});
 };
 
-module.exports = mongoose.model('User', userSchema);
+/**
+ * Read and Retrieve a User object from the database
+ * @param {string} user's googleID (as hash)
+ * @returns {User} one User object with matching ID
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => console.log(user))
+ *     .catch(error => console.error(error));
+ */
+
+userSchema.statics.read = function(googleId){
+	return new Promise((resolve , reject) => {
+		User.findOne({googleID: googleId}, (err, user) => {
+			if (err) reject(err);
+			else resolve(user);
+		});
+	});
+};
+
+/**
+ * Delete a User object from the database
+ * @param {string} user's googleID (as hash)
+ * @returns {void} nothing
+ * @example
+ * User.delete('someGoogleIdHash')
+ *     .catch(error => console.error(error));
+ */
+
+userSchema.statics.delete = function(googleId){
+	return new Promise((resolve, reject) => {
+		User.findOne({googleID: googleId}).remove((err, offer) => {
+			if(err) reject(err);
+			else resolve(offer);
+		});
+	});
+};
+
+/**
+ * Change User's isAdmin status
+ * @param {Boolean} User's new isAdmin status
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.setAdminStatus(true);
+ *     })
+ *	   .catch(error => console.error(error));
+ */
+
+userSchema.methods.setAdminStatus = function(adminStatus){
+	this.isAdmin = adminStatus;
+	
+	return new Promise((resolve, reject) => {
+		this.save((error, user) => {
+			if(error) reject(error);
+			else resolve(user);
+		});
+	});
+};
+
+/**
+ * Change User's isMentor status
+ * @param {Boolean} User's new isMentor status
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.setMentorStatus(true);
+ *     })
+ *	   .catch(error => console.error(error));
+ */
+
+userSchema.methods.setMentorStatus = function(mentorStatus){
+	this.isMentor = mentorStatus;
+
+	return new Promise((resolve, reject) => {
+		this.save((err, user) => {
+			if(err) reject(err);
+			else resolve(user);
+		});
+	});
+};
+
+/**
+ * Add a new skill to the User object
+ * @param {string} User's new skill
+ * @returns {User} changed user object
+ * @example
+ * User.read('someGoogleIdHash')
+ *     .then(user => {
+ *	        user.addSkill('python');
+ *     })
+ *	   .catch(error => console.error(error));
+ *
+ * If attempt to add a skill that already exists, 
```

Comment:
Put this additional description not part of the example, but with the description at the top of this commented block